src/gsi/get_gefs_for_regional.f90

```diff
@@ -336,16 +354,117 @@ subroutine get_gefs_for_regional
+        !sfcges = open_dataset(sfilename)
+        write(6,*) ' LB netcdf test2'
+        ! get dimension sizes
+        ncdim = get_dim(atmges, 'grid_xt'); gfshead%lonb = ncdim%len
+        ncdim = get_dim(atmges, 'grid_yt'); gfshead%latb = ncdim%len
+        ncdim = get_dim(atmges, 'pfull') ; gfshead%levs = ncdim%len
+        write(6,*) 'region lonb latb=',gfshead%lonb, gfshead%latb,gfshead%levs         
+        ! LB: add nsig_gfs here: 
+        nsig_gfs = gfshead%levs
+        ! hard code jcap,idsl,idvc
+        gfshead%jcap = -9999
+        gfshead%idsl= 1
+        gfshead%idvc = 2
+
+        ! FV3GFS write component does not include JCAP, infer from DIMY-2
+        if (njcap<0) njcap=latb-2
+
+        nlat_gfs=gfshead%latb+2
+        nlon_gfs=gfshead%lonb
+        nsig_gfs=gfshead%levs
+        write(6,*) 'jcap', jcap_ens
+        write(6,*) 'jcap', gfshead%jcap
+        write(6,*) 'jcap1'
+
+!         jcap_gfs=latb-2
+         jcap_gfs=gfshead%latb-2
+
+!        if(gfshead%jcap > 0)then
+!          jcap_gfs=gfshead%jcap
+!        else if(jcap_ens > 0)then
+!          jcap_gfs=jcap_ens
+!        else
+!          write(6,*)'get_gefs_for_regional:ERROR ncio jcap is undefined'
+!        endif
+
+        write(6,*) 'regional jcap_gfs=',jcap_gfs
+
+
+!        call read_attribute(atmges, 'ncnsto', ntrac)
+!        gfshead%ntrac = ntrac(1)
+!        call read_attribute(sfcges, 'ncld', ncld)
+!        gfshead%ncldt = ncld(1)
+
+!        write(6,*) 'regional ntrac=', gfshead%ntrac
+!        write(6,*) 'regional ncldt=', gfshead%ncldt
+
+!        call close_dataset(sfcges)
+        if (mype==mype_out) write(6,*)'GESINFO:  Read NCEP FV3GFS netCDF ', &
+           'format file, ',trim(filename)
+        ! hard code nvcoord to be 2
+        gfshead%nvcoord=2 ! ak and bk
+        write(6,*) 'nvcoord=', gfshead%nvcoord
+        if (allocated(gfsheadv%vcoord)) deallocate(gfsheadv%vcoord)
+        allocate(gfsheadv%vcoord(gfshead%levs+1,gfshead%nvcoord))
+        write(6,*) 'test1'
+        call read_attribute(atmges, 'ak', aknc)
+        call read_attribute(atmges, 'bk', bknc)
+        write(6,*) 'test2'
+        write(6,*) 'ak,bk=', aknc, bknc
+        do k=1,gfshead%levs+1
+           kr = gfshead%levs+2-k
+           gfsheadv%vcoord(k,1) = aknc(kr)
+           gfsheadv%vcoord(k,2) = bknc(kr)
+        end do
+        deallocate(aknc,bknc) 
+        ! get time information
+        idate6 = get_idate_from_time_units(atmges)
+        gfshead%idate(1) = idate6(4)  !hour
+        gfshead%idate(2) = idate6(2)  !month
+        gfshead%idate(3) = idate6(3)  !day
+        gfshead%idate(4) = idate6(1)  !year
+        call read_vardata(atmges, 'time', fhour1) ! might need to change this to attribute later
+                                               ! depends on model changes from Jeff Whitaker
+        gfshead%fhour = fhour1(1)
+
+        call close_dataset(atmges)
+
+        write(6,*) ' netCDF:fhour,idate=',fhour1,idate6
```

Comment:
These print statements should be wrapped in an if statement to check for mype, otherwise it will print too much to stdout for N MPI tasks.

Reply:
Added this in an if  statement to check for mype.